hack/generated/controllers/crd_networking_loadbalancer_test.go

```diff
@@ -0,0 +1,124 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	network "github.com/Azure/azure-service-operator/hack/generated/_apis/microsoft.network/v1alpha1api20200501"
+	"github.com/Azure/azure-service-operator/hack/generated/pkg/genruntime"
+	"github.com/Azure/azure-service-operator/hack/generated/pkg/testcommon"
+)
+
+func Test_LoadBalancer_CRUD(t *testing.T) {
+	t.Parallel()
+
+	g := NewGomegaWithT(t)
+	ctx := context.Background()
+	testContext, err := testContext.ForTest(t)
+	g.Expect(err).ToNot(HaveOccurred())
+	rg, err := testContext.CreateNewTestResourceGroup(testcommon.WaitForCreation)
+	g.Expect(err).ToNot(HaveOccurred())
+
+	// Public IP Address
+	sku := network.PublicIPAddressSkuNameStandard
+	publicIPAddress := &network.PublicIPAddresses{
+		TypeMeta: metav1.TypeMeta{
+			Kind: reflect.TypeOf(network.PublicIPAddresses{}).Name(),
+		},
+		ObjectMeta: testContext.MakeObjectMetaWithName(testContext.Namer.GenerateName("publicip")),
+		Spec: network.PublicIPAddresses_Spec{
+			Location: testContext.AzureRegion,
+			Owner:    testcommon.AsOwner(rg.ObjectMeta),
+			Sku: &network.PublicIPAddressSku{
+				Name: &sku,
+			},
+			Properties: network.PublicIPAddressPropertiesFormat{
+				PublicIPAllocationMethod: network.PublicIPAddressPropertiesFormatPublicIPAllocationMethodStatic,
+			},
+		},
+	}
+
+	err = testContext.KubeClient.Create(ctx, publicIPAddress)
+	g.Expect(err).ToNot(HaveOccurred())
+	// It should be created in Kubernetes
+	g.Eventually(publicIPAddress).Should(testContext.Match.BeProvisioned(ctx))
+	g.Expect(publicIPAddress.Status.Id).ToNot(BeNil())
+
+	// LoadBalancer
+	loadBalancerSku := network.LoadBalancerSkuNameStandard
+	lbName := testContext.Namer.GenerateName("loadbalancer")
+	lbFrontendName := "LoadBalancerFrontend"
+	loadBalancer := &network.LoadBalancer{
+		ObjectMeta: testContext.MakeObjectMetaWithName(lbName),
+		Spec: network.LoadBalancers_Spec{
+			Location: testContext.AzureRegion,
+			Owner:    testcommon.AsOwner(rg.ObjectMeta),
+			Sku: &network.LoadBalancerSku{
+				Name: &loadBalancerSku,
+			},
+			Properties: network.LoadBalancerPropertiesFormat{
+				FrontendIPConfigurations: []network.FrontendIPConfiguration{
+					{
+						Name: lbFrontendName,
+						Properties: &network.FrontendIPConfigurationPropertiesFormat{
+							PublicIPAddress: &network.SubResource{
+								Reference: testContext.MakeReferenceFromResource(publicIPAddress),
+							},
+						},
+					},
+				},
+				// TODO: The below stuff isn't really necessary for LB CRUD but is required for VMSS...
+				InboundNatPools: []network.InboundNatPool{
+					{
+						Name: "MyFancyNatPool",
+						Properties: &network.InboundNatPoolPropertiesFormat{
+							FrontendIPConfiguration: network.SubResource{
+								Reference: genruntime.ResourceReference{
+									// TODO: This is still really awkward
+									ARMID: testContext.MakeARMId(rg.Name, "Microsoft.Network", "loadBalancers", lbName, "frontendIPConfigurations", lbFrontendName),
```

Comment:
This is _really_ awkward. We may want to find a way to work around this with an admissions webhook or custom hook in this controller...

Reply:
@matthchr we should be able to build this automatically from the Swagger specs

Reply:
@porges can you expand? This includes a reference to a different resource that:

1. Isn't _really_ a resource (it's a resource created by a property above).
2. Doesn't exist yet and whose name is variable (based on the spec of this resource)

Reply:
@matthchr, not that it will be automatic, but e.g. the Swagger spec has:

```
/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/loadBalancers/{loadBalancerName}/frontendIPConfigurations/{frontendIPConfigurationName}
```

However, since as you say it’s not _really_ a resource we won’t have anything to attach it to…